examples/player/player_wrapper.rs

```diff
@@ -70,53 +71,76 @@ impl PlayerWrapper {
         let servo_media = ServoMedia::get().unwrap();
         let player = Arc::new(Mutex::new(servo_media.create_player(StreamType::Seekable)));
+
         let use_gl = if let Some(windowed_context) = windowed_context {
             PlayerWrapper::set_gl_params(&player, windowed_context).is_ok()
         } else {
             false
         };
+
         let file = File::open(&path).unwrap();
         let metadata = file.metadata().unwrap();
         player
             .lock()
             .unwrap()
             .set_input_size(metadata.len())
             .unwrap();
+
         let (sender, receiver) = ipc::channel().unwrap();
         player.lock().unwrap().register_event_handler(sender);
+
         let player_ = player.clone();
         let player__ = player.clone();
         let shutdown = Arc::new(AtomicBool::new(false));
         let shutdown_ = shutdown.clone();
         let shutdown__ = shutdown.clone();
+
+        let (seek_sender, seek_receiver) = mpsc::channel();
+
         Builder::new()
             .name("File reader".to_owned())
             .spawn(move || {
                 let player = &player_;
                 let shutdown = &shutdown_;
+
                 let mut buf_reader = BufReader::new(file);
                 let mut buffer = [0; 8192];
+                let end_file = AtomicBool::new(false);
+
                 while !shutdown.load(Ordering::Relaxed) {
-                    match buf_reader.read(&mut buffer[..]) {
-                        Ok(0) => {
-                            println!("finished pushing data");
+                    if let Ok(offset) = seek_receiver.try_recv() {
+                        if buf_reader.seek(SeekFrom::Start(offset)).is_err() {
+                            eprintln!("BufReader - Could not seek to {:?}", offset);
                             break;
                         }
-                        Ok(size) => {
-                            if let Err(_) = player
-                                .lock()
-                                .unwrap()
-                                .push_data(Vec::from(&buffer[0..size]))
-                            {
+                        end_file.store(false, Ordering::Relaxed);
+                    }
+
+                    if !end_file.load(Ordering::Relaxed) {
+                        match buf_reader.read(&mut buffer[..]) {
+                            Ok(0) => {
+                                println!("finished pushing data");
+                                end_file.store(true, Ordering::Relaxed);
+                            }
+                            Ok(size) => {
+                                if let Err(e) = player
+                                    .lock()
+                                    .unwrap()
+                                    .push_data(Vec::from(&buffer[0..size]))
+                                {
+                                    println!("Can't push data: {:?}", e);
+                                    //break;
```

Comment:
Remove this comment?

Reply:
Actually it is a hack left behind. Let bring back the break if it is not a enough data.